### ISSUE TITLE
Remove farReuse?

### ISSUE THREAD

**Issue by nad** [agda/agda#4471]
The `farReuse` field is always updated by the serialiser, even if no `-v` flag has been given:

https://github.com/agda/agda/blob/8c21b6be0c98a7c87d0c65ac5c9242967b0e11b7/src/full/Agda/TypeChecking/Serialise/Base.hs#L57-L62

Furthermore the `farReuse` field for nodes is updated *many* times. I suggest that we remove the `farReuse` field unless Agda is compiled with the debug flag turned on.

**Comment by andreasabel** [comment]
I'd be surprised if these counters take a noticeable share of the cost of serialization.

**Comment by nad** [comment]
I ran the library test once with these counters, and once without. The serialisation time decreased from about 74 s to about 72 s. This might be within the margin of error for such a measurement. Would you object if I push the changes?

**Comment by andreasabel** [comment]
Make a PR!

**Comment by nad** [comment]
Why? The question is whether you would object to defining `FreshAndReuse` in the following way:
```haskell
#ifdef DEBUG
data FreshAndReuse = FreshAndReuse
  { farFresh :: !Int32 -- ^ Number of hash map misses.
  , farReuse :: !Int32 -- ^ Number of hash map hits.
  }
#else
newtype FreshAndReuse = FreshAndReuse
  { farFresh :: Int32 -- ^ Number of hash map misses.
  }
#endif
```

**Comment by andreasabel** [comment]
No, I wouldn't.  Please go ahead.
A question is whether we ever test compilation with `DEBUG` on, to prevent bit-rotting conditional code...

**Comment by asr** [comment]
> A question is whether we ever test compilation with `DEBUG` on, to prevent bit-rotting conditional code...

I added this test some time ago. The current version of this test lives in

https://github.com/agda/agda/blob/af74e45ed2177bed902c86910d4795ca90e20173/.github/workflows/stack.yml#L28

**Comment by asr** [comment]
Merged into the `release-2.6.1` branch.
